src/main/java/seedu/savenus/logic/parser/SortCommandParser.java

```diff
@@ -0,0 +1,123 @@
+package seedu.savenus.logic.parser;
+
+import static seedu.savenus.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.savenus.logic.commands.SortCommand;
+import seedu.savenus.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new SortCommand object.
+ */
+public class SortCommandParser implements Parser<SortCommand> {
+    /**
+     * Parses the given String and returns a SortCommand object for simple execution.
+     * @throws ParseException if the String contains duplicate or invalid fields.
+     */
+    public SortCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.NO_ARGUMENTS_USAGE));
+        }
+
+        String[] nameKeywords = trimmedArgs.split("\\s+");
+        if (areFieldsInvalid(nameKeywords)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.INVALID_FIELD_USAGE));
+        } else if (areFieldsDuplicate(nameKeywords)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.DUPLICATE_FIELD_USAGE));
+        } else if (areDirectionsMissing(nameKeywords)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MISSING_DIRECTION_USAGE));
+        } else if (areDirectionsInaccurate(nameKeywords)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.INVALID_DIRECTION_USAGE));
+        }
+
+        return new SortCommand((Arrays.asList(nameKeywords)));
+    }
+
+    /**
+     * Checks if the array of fields contain invalid values.
+     * @param keywords the array of fields.
+     * @return true if the fields do contain invalid fields. False if otherwise.
+     */
+    public boolean areFieldsInvalid(String[] keywords) {
+        for (int i = 0; i < keywords.length; i = i + 2) {
+            String field = keywords[i];
+            if (!isValidField(field)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if the directions are missing for certain fields.
+     * @param keywords the array of fields.
+     * @return true if directions are missing, false if otherwise.
+     */
+    public boolean areDirectionsMissing(String[] keywords) {
+        return keywords.length % 2 != 0;
+    }
```

Comment:
You might want to make this more specific? incase got some user very creative.
Just my opinion but otherwise good enough tbh

Reply:
There will be checks after this so no worries 😉